ios/keyman/Keyman/Keyman/MainViewController.swift

```diff
@@ -0,0 +1,1200 @@
+      let kbDict = dictionaryWithKeyboardID(kbID, languageID: langID)
+      let languageName = kbDict?[kKeymanLanguageNameKey] ?? ""
+      let title = "\(languageName) Font"
+      let msg = "Touch Install to make \(languageName) display correctly in all your apps"
+      showAlert(withTitle: title, message: msg, cancelButtonTitle: "Cancel", otherButtonTitles: "Install", tag: 1)
+    } else {
+      profileName = nil
+    }
+  }
+
+  func showKeyboard() {
+    textView.becomeFirstResponder()
+  }
+
+  func showAlert(withTitle title: String, message msg: String, cancelButtonTitle cbTitle: String?,
+                 otherButtonTitles obTitles: String?, tag: Int) {
+    dismissGetStartedView(nil)
+    let alertView = UIAlertView(title: title, message: msg, delegate: self, cancelButtonTitle: cbTitle,
+                                otherButtonTitles: obTitles ?? "")
+    alertView.tag = tag
+    alertView.show()
+  }
+
+  func performAlertButtonClick(withTag tag: Int) {
+    switch tag {
+    case 0:
+      let langID = keyboard2Download[kKeymanLanguageIdKey] as? String
+      let kbID = keyboard2Download[kKeymanKeyboardIdKey] as? String
+      KMManager.sharedInstance().downloadKeyboard(withID: kbID, languageID: langID, isUpdate: false)
+    case 1:
+      if let profileName = profileName {
+        checkedProfiles.append(profileName)
+        let userData = AppDelegate.activeUserDefaults()
+        userData.set(checkedProfiles, forKey: checkedProfilesKey)
+        userData.synchronize()
+
+        UIApplication.shared.openURL(URL(string: "\(baseUri)\(profileName)")!)
+        self.profileName = nil
+      }
+    case 2:
+      if let jsonUrl = customKeyboard2Download["url"] as? URL,
+        let isDirect = customKeyboard2Download["direct"] as? Bool {
+        KMManager.sharedInstance().downloadKeyboard(from: jsonUrl, isDirect: isDirect)
+      }
+    default:
+      break
+    }
+  }
+
+  func alertView(_ alertView: UIAlertView, clickedButtonAt buttonIndex: Int) {
+    if buttonIndex != alertView.cancelButtonIndex {
+      performAlertButtonClick(withTag: alertView.tag)
+    }
+
+    if let profileName = profileName, alertView.tag == 1 {
+      checkedProfiles.append(profileName)
+      let userData = AppDelegate.activeUserDefaults()
+      userData.set(checkedProfiles, forKey: checkedProfilesKey)
+      userData.synchronize()
+      self.profileName = nil
+    } else if alertView.tag == 2 {
+      if shouldShowGetStarted() {
+        showGetStartedView(nil)
+      }
+    }
+  }
+
+  func showDropDownMenu(_ sender: Any) {
+    if dismissDropDownMenu() {
+      return
+    }
+
+    wasKeyboardVisible = textView.isFirstResponder
+    textView.dismissKeyboard()
+
+    let w: CGFloat = 160
+    let h: CGFloat = 44
+    let x: CGFloat = navBarWidth() - w
+    let y: CGFloat = AppDelegate.statusBarHeight() + navBarHeight()
+
+    let dropDownList = DropDownListView(listItems: dropdownItems, itemSize: CGSize(width: w, height: h),
```

Comment:
`dropdownItems` used to be an inline-defined array in the Obj-C code, but I don't see its equivalent in this method.  In fact, where is this array even defined now?  How is the method using it?

Reply:
It seems to be defined within the function `setNavBarButtons()` - but that would make it a local variable, no?  There's no class variable with that name defined in a way I recognize.  Or is this managed somehow elsewhere?

Reply:
`dropdownItems` is declared here:
https://github.com/keymanapp/keyman/blob/136e08b2f1c66ba856443ec9c976ba3e7c22cc71/ios/keyman/Keyman/Keyman/MainViewController.swift#L38

Setting an array instead of four different buttons just makes the nil checking easier.